controller/src/main/java/io/pravega/controller/server/Main.java

```diff
@@ -86,10 +90,14 @@ public static void main(String[] args) {
             controllerServiceMain.startAsync();
             controllerServiceMain.awaitTerminated();
 
+            statsProvider.close();
             log.info("Controller service exited");
             System.exit(0);
         } catch (Throwable e) {
             log.error("Controller service failed", e);
+            if (statsProvider != null) {
```

Comment:
Put this in the `finally` block; that way you only need to write it once.

Reply:
Done.